src/hooks/useFetchOnMount.ts

```diff
@@ -0,0 +1,23 @@
+import { useEffect, useState } from "react";
+
+export function useFetchOnMount<T>(apiEndpoint: string) {
+  const [data, updateData] = useState<T>();
+
+  useEffect(() => {
+    const abortController = new AbortController();
+
+    async function fetchData() {
+      const response = await fetch(apiEndpoint, {signal: abortController.signal});
+      const data = await response.json();
+      updateData(data);
+    }
+
+    fetchData();
+
+    return () => {
+      abortController.abort();
+    }
+  })
```

Comment:
Ooops! Looks like there's a bug in here. Can you find it?

Reply:
I believe that on line 18 the method needs to be .sgnal
```suggestion
  useEffect(() => {
    const abortController = new AbortController();

    async function fetchData() {
      const response = await fetch(apiEndpoint, {signal: abortController.signal});
      const data = await response.json();
      updateData(data);
    }

    fetchData();

    return () => {
      abortController.signal();
    }
  })
```

Reply:
because we want this effect to only be used on the first render, we need to add a empty array as the second argument to useEffect, otherwise we will get a memory leak!

```suggestion
  useEffect(() => {
    const abortController = new AbortController();

    async function fetchData() {
      const response = await fetch(apiEndpoint, {signal: abortController.signal});
      const data = await response.json();
      updateData(data);
    }

    fetchData();

    return () => {
      abortController.abort();
    }
  })
```